### PR TITLE
Feature/follow-users

### PR DESCRIPTION
**Pull request by lui7henrique**
## Describe your changes
This pull request implements a feature to follow users through their profile, or in the user's list.

In addition to reducing the number of static pages to reduce construction time, while still keeping these pages in the sitemap.

## Issue ticket number and link
Fixes PLO-32 (#139)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it's an essential feature, I've tested it thoroughly.
- [x] Do we need to implement analytics?
- [x] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

File: apps/web/src/components/follow-button/follow-button.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const { user } = useAuth()
+  const { dictionary } = useLanguage()
+  const { handleFollow, handleRemoveFollow } = useFollowers()
+
+  const { data: followers, isLoading } = useQuery({
+    queryKey: ['followers', profileId],
+    queryFn: async () => await getFollowers(profileId),
+  })
+
+  const userFollow = followers?.find((f) => f.follower_id === user?.id)
```
Comment (vmnog) [inline apps/web/src/components/follow-button/follow-button.tsx:22]: Maybe rename userFollow to isFollower 

```typescript
const isFollower = followers?.find((f) => f.follower_id === user?.id)

File: apps/web/src/components/follow-button/follow-button.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  const getMode = () => {
```
Comment (vmnog) [inline apps/web/src/components/follow-button/follow-button.tsx:24]: Maybe rename getMode to getFollowButton
```typescript
const getFollowButton= () => {

**Review by vmnog — APPROVED** [review]
I suggested some rename on variables to make it clear, but overall its ok to merge this. Great job!

File: apps/web/src/components/follow-button/follow-button.tsx (hunk 1 above)
Comment (lui7henrique) [reply apps/web/src/components/follow-button/follow-button.tsx:22]: Thiis does not return a boolean. Returns the user's follow if there is one, and if so, the follow id is used to unfollow

**Pull request merged**